server/types/text.go

```diff
@@ -106,6 +106,10 @@ func (b TextType) Convert(val any) (any, sql.ConvertInRange, error) {
 
 // Equals implements the DoltgresType interface.
 func (b TextType) Equals(otherType sql.Type) bool {
+	if _, ok := otherType.(TextType); !ok {
```

Comment:
I'm assuming this change was made because the existing `Equals` function was not properly catching the appropriate cases, which would be fairly concerning. What's the reason for it? If it's something to do with parameter resolution, then we need to get that fixed as soon as we can.

Reply:
The `text` overload gets compared to the `regclass` overload, but `regclass` doesn't implement `Serialize` and throws a panic. So, nothing to do with parameter resolution per-se, just that `regclass` cannot be compared that way to any other types.